app/post_metadata.go

```diff
@@ -123,6 +123,15 @@ func (a *App) PreparePostForClient(c request.CTX, originalPost *model.Post, isNe
 		post.Metadata.Files = fileInfos
 	}
 
+	if a.Config().FeatureFlags.PostPriority && *a.Config().ServiceSettings.PostPriority && post.RootId == "" {
+		// Post's Priority if any
+		if priority, err := a.GetPriorityForPost(post.Id); err != nil {
+			mlog.Warn("Failed to get post priority for a post", mlog.String("post_id", post.Id), mlog.Err(err))
+		} else {
+			post.Metadata.Priority = priority
+		}
+	}
+
```

Comment:
Would I be correct in tracing the code to see `App.PreparePostListForClient` effectively calling this method in a loop for a set of posts, e.g. when fetching root posts in a channel? Do we need a means to fetch this priority in bulk instead?

Reply:
I've made a try on this, so when requesting PreparePostListForClient priority metadata are fetched in bulk, but we also need to, optionally, add priority when we prepare a single post for the client (PreparePostForClient)